src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -1645,5 +1645,173 @@ internal static Exception CreateInvalidNameCharException(string name, int index,
+        {
+            charsWritten = 1;
+            if (destination.Length < 1) return false;
+
+            destination[0] = value;
+            return true;
+        }
+
+        internal static bool TryFormat(decimal value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, NumberFormatInfo.InvariantInfo);
+        }
+
+        internal static bool TryFormat(sbyte value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out  charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(short value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(int value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(long value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(byte value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(ushort value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(uint value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(ulong value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(float value, Span<char> destination, out int charsWritten)
+        {
+            ReadOnlySpan<char> valueSpan = default;
+
+            if (float.IsNegativeInfinity(value))
+            {
+                valueSpan = "-INF";
+            }
+            else if (float.IsPositiveInfinity(value))
+            {
+                valueSpan = "INF";
+            }
+            else if (IsNegativeZero((double)value))
+            {
+                valueSpan = "-0";
+            }
+
+            if (valueSpan.IsEmpty)
+                return value.TryFormat(destination, out charsWritten, "R", NumberFormatInfo.InvariantInfo);
```

Comment:
If you place this in an else block, you avoid an extra conditional check.

Reply:
done